File: address-geocode-canada-python/REST/agca_response.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class GetGeoLocationInput:
    Address: Optional[str] = None
    Municipality: Optional[str] = None
    Province: Optional[str] = None
    PostalCode: Optional[str] = None
    LicenseKey: Optional[str] = None
    IsLive: bool = True
    TimeoutSeconds: int = 15

    def __str__(self) -> str:
        return (f"GetGeoLocationInput: Address={self.Address}, Municipality={self.Municipality}, "
                f"Province={self.Province}, PostalCode={self.PostalCode}, LicenseKey={self.LicenseKey}, "
                f"IsLive={self.IsLive}, TimeoutSeconds={self.TimeoutSeconds}")

@dataclass
class GetPostalCodeInfoInput:
    PostalCode: Optional[str] = None
    LicenseKey: Optional[str] = None
    IsLive: bool = True
    TimeoutSeconds: int = 15

    def __str__(self) -> str:
        return (f"GetPostalCodeInfoInput: PostalCode={self.PostalCode}, LicenseKey={self.LicenseKey}, "
                f"IsLive={self.IsLive}, TimeoutSeconds={self.TimeoutSeconds}")

@dataclass
class GetGeoLocationByMunicipalityProvinceInput:
    Municipality: Optional[str] = None
    Province: Optional[str] = None
    LicenseKey: Optional[str] = None
    IsLive: bool = True
    TimeoutSeconds: int = 15

    def __str__(self) -> str:
        return (f"GetGeoLocationByMunicipalityProvinceInput: Municipality={self.Municipality}, "
                f"Province={self.Province}, LicenseKey={self.LicenseKey}, "
                f"IsLive={self.IsLive}, TimeoutSeconds={self.TimeoutSeconds}")

@dataclass
class Error:
    Desc: Optional[str] = None
    Number: Optional[str] = None
    Location: Optional[str] = None

    def __str__(self) -> str:
        return f"Error: Desc={self.Desc}, Number={self.Number}, Location={self.Location}"

@dataclass
class GetGeoLocationResponse:
    Latitude: Optional[str] = None
    Longitude: Optional[str] = None
    PostalCode: Optional[str] = None
    MatchCode: Optional[str] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        error = str(self.Error) if self.Error else 'None'
        return (f"GetGeoLocationResponse: Latitude={self.Latitude}, Longitude={self.Longitude}, "
                f"PostalCode={self.PostalCode}, MatchCode={self.MatchCode}, Error={error}")

@dataclass
class GetPostalCodeInfoResponse:
    Latitude: Optional[str] = None
    Longitude: Optional[str] = None
    PostalCode: Optional[str] = None
    TimeZone: Optional[str] = None
    DST: Optional[str] = None
    AreaCode: Optional[str] = None
    City: Optional[str] = None
    CityPopulation: Optional[str] = None
    Province: Optional[str] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        error = str(self.Error) if self.Error else 'None'
        return (f"GetPostalCodeInfoResponse: Latitude={self.Latitude}, Longitude={self.Longitude}, "
                f"PostalCode={self.PostalCode}, TimeZone={self.TimeZone}, DST={self.DST}, "
                f"AreaCode={self.AreaCode}, City={self.City}, CityPopulation={self.CityPopulation}, "
                f"Province={self.Province}, Error={error}")

@dataclass
class GetGeoLocationByMunicipalityProvinceResponse:
    Latitude: Optional[str] = None
    Longitude: Optional[str] = None
    PostalCode: Optional[str] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        error = str(self.Error) if self.Error else 'None'
        return (f"GetGeoLocationByMunicipalityProvinceResponse: Latitude={self.Latitude}, "
                f"Longitude={self.Longitude}, PostalCode={self.PostalCode}, Error={error}")

File: address-geocode-canada-nodejs/REST/agca_response.js
/**
 * Input parameters for the GetGeoLocation API call.
 */
export class GetGeoLocationInput {
    constructor(data = {}) {
        this.Address = data.Address;
        this.Municipality = data.Municipality;
        this.Province = data.Province;
        this.PostalCode = data.PostalCode;
        this.LicenseKey = data.LicenseKey;
        this.IsLive = data.IsLive !== undefined ? data.IsLive : true;
        this.TimeoutSeconds = data.TimeoutSeconds !== undefined ? data.TimeoutSeconds : 15;
    }

    toString() {
        return `GetGeoLocationInput: Address = ${this.Address}, Municipality = ${this.Municipality}, Province = ${this.Province}, PostalCode = ${this.PostalCode}, LicenseKey = ${this.LicenseKey}, IsLive = ${this.IsLive}, TimeoutSeconds = ${this.TimeoutSeconds}`;
    }
}

/**
 * Input parameters for the GetPostalCodeInfo API call.
 */
export class GetPostalCodeInfoInput {
    constructor(data = {}) {
        this.PostalCode = data.PostalCode;
        this.LicenseKey = data.LicenseKey;
        this.IsLive = data.IsLive !== undefined ? data.IsLive : true;
        this.TimeoutSeconds = data.TimeoutSeconds !== undefined ? data.TimeoutSeconds : 15;
    }

    toString() {
        return `GetPostalCodeInfoInput: PostalCode = ${this.PostalCode}, LicenseKey = ${this.LicenseKey}, IsLive = ${this.IsLive}, TimeoutSeconds = ${this.TimeoutSeconds}`;
    }
}

/**
 * Input parameters for the GetGeoLocationByMunicipalityProvince API call.
 */
export class GetGeoLocationByMunicipalityProvinceInput {
    constructor(data = {}) {
        this.Municipality = data.Municipality;
        this.Province = data.Province;
        this.LicenseKey = data.LicenseKey;
        this.IsLive = data.IsLive !== undefined ? data.IsLive : true;
        this.TimeoutSeconds = data.TimeoutSeconds !== undefined ? data.TimeoutSeconds : 15;
    }

    toString() {
        return `GetGeoLocationByMunicipalityProvinceInput: Municipality = ${this.Municipality}, Province = ${this.Province}, LicenseKey = ${this.LicenseKey}, IsLive = ${this.IsLive}, TimeoutSeconds = ${this.TimeoutSeconds}`;
    }
}

/**
 * Error object for API responses.
 */
export class Error {
    constructor(data = {}) {
        this.Desc = data.Desc;
        this.Number = data.Number;
        this.Location = data.Location;
    }

    toString() {
        return `Error: Desc = ${this.Desc}, Number = ${this.Number}, Location = ${this.Location}`;
    }
}

/**
 * Response from GetGeoLocation API, containing the latitude, longitude, postal code, 
 * and match code for the given Canadian address.
 */
export class GetGeoLocationResponse {
    constructor(data = {}) {
        this.Latitude = data.Latitude;
        this.Longitude = data.Longitude;
        this.PostalCode = data.PostalCode;
        this.MatchCode = data.MatchCode;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `GetGeoLocationResponse: Latitude = ${this.Latitude}, Longitude = ${this.Longitude}, PostalCode = ${this.PostalCode}, MatchCode = ${this.MatchCode}, Error = ${this.Error ? this.Error.toString() : 'null'}`;
    }
}

/**
 * Response from GetPostalCodeInfo API, containing the latitude, longitude, postal code, 
 * time zone, DST, area code, city, city population, province, and error for the given Canadian postal code.
 */
export class GetPostalCodeInfoResponse {
    constructor(data = {}) {
        this.Latitude = data.Latitude;
        this.Longitude = data.Longitude;
        this.PostalCode = data.PostalCode;
        this.TimeZone = data.TimeZone;
        this.DST = data.DST;
        this.AreaCode = data.AreaCode;
        this.City = data.City;
        this.CityPopulation = data.CityPopulation;
        this.Province = data.Province;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `GetPostalCodeInfoResponse: Latitude = ${this.Latitude}, Longitude = ${this.Longitude}, PostalCode = ${this.PostalCode}, TimeZone = ${this.TimeZone}, DST = ${this.DST}, AreaCode = ${this.AreaCode}, City = ${this.City}, CityPopulation = ${this.CityPopulation}, Province = ${this.Province}, Error = ${this.Error ? this.Error.toString() : 'null'}`;
    }
}

/**
 * Response from GetGeoLocationByMunicipalityProvince API, containing the latitude, longitude, 
 * postal code, and error for the given Canadian municipality and province.
 */
export class GetGeoLocationByMunicipalityProvinceResponse {
    constructor(data = {}) {
        this.Latitude = data.Latitude;
        this.Longitude = data.Longitude;
        this.PostalCode = data.PostalCode;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `GetGeoLocationByMunicipalityProvinceResponse: Latitude = ${this.Latitude}, Longitude = ${this.Longitude}, PostalCode = ${this.PostalCode}, Error = ${this.Error ? this.Error.toString() : 'null'}`;
    }
}

export default GetGeoLocationResponse;